mcs/class/System/Test/System/UriTest3.cs

```diff
@@ -696,6 +696,39 @@ public void DomainLabelLength ()
 			type = Uri.CheckHostName ("3.141592653589793238462643383279502884197169399375105820974944592._om");
 			Assert.AreEqual (UriHostNameType.Unknown, type, "DomainLabelLength#3");
 		}
+
+        // https://bugzilla.xamarin.com/show_bug.cgi?id=58400
+        [Test]
+        public static void Test_LocalPath_Bug58400()
+        {
+            var uriAndExpected = new global::System.Collections.Generic.Dictionary<string, string> 
```

Comment:
Could be just an array of named tuples

Reply:
Would be more code, wouldn't it? Because of the new Tuple<,>() per item

Reply:
C# supports e.g. `(value: string, data: string)` syntax for tuples